iroh-net/src/netcheck/reportgen.rs

```diff
@@ -881,9 +887,23 @@ async fn check_captive_portal(dm: &RelayMap, preferred_relay: Option<RelayUrl>)
         }
     };
 
-    let client = reqwest::ClientBuilder::new()
-        .redirect(reqwest::redirect::Policy::none())
-        .build()?;
+    let mut builder = reqwest::ClientBuilder::new().redirect(reqwest::redirect::Policy::none());
+    if let Some(Host::Domain(domain)) = url.host() {
+        // Use our own resolver rather than getaddrinfo
+        //
+        // For some reason reqwest wants SocketAddr rather than IpAddr but then proceeds to
+        // ignore the port, extracting it from the URL instead.  We supply a dummy port.
+        //
+        // Ideally we would try to resolve **both** IPv4 and IPv6 rather than purely race
+        // them.  But our resolver doesn't support that yet.
+        let addrs: Vec<_> = dns_resolver
+            .lookup_ipv4_ipv6_staggered(domain, DNS_TIMEOUT, DNS_STAGGERING_MS)
+            .await?
+            .map(|ipaddr| SocketAddr::new(ipaddr, 80))
```

Comment:
@flub I think you should probably set these ports to 0, if I'm understanding [this issue](https://github.com/seanmonstar/reqwest/issues/2413#issuecomment-2405545067) correctly.

Reply:
The docs have seemingly not been updated yet. I'll look into the source if they're actually unused or not now.

Reply:
Good catch!  Also, bad reqwest!  https://github.com/n0-computer/iroh/pull/2906

Reply:
Sorry... the change was in `hyper_util` not in reqwest so that is why the documentation stayed outdated. https://github.com/seanmonstar/reqwest/pull/2473

Reply:
In my defense, this change in `hyper_util` is really useful to leverage HTTPS records in custom resolvers. Even more work needed on Reqwest side to decide when to switch to HTTP2/3 according to the ALPN property in HTTPS records.

Reply:
All good. IMO it's more weird to pass in bogus info that's unused. This is what made me skeptical in the first place.

Reply:
Also ugh I didn't quite follow the data-flow in reqwest for how these domain overrides are used. It goes through some `Resolver` and `tower` traits and just gets complicated quickly. So I'm glad to see someone who knows just update the docs.